tools/generate-test-case

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/python3
+
+import argparse
+import subprocess
+import json
+import os
+import sys
+
+'''
+This script generates a json test case. It accepts a test_case_request as input through standard input.
+
+{
+    "boot": {
+        "type": "qemu"
+    },
+    "compose-request": {
+        "distro": "fedora-30",
+        "arch": "x86_64",
+        "output-format": "qcow2",
+        "filename": "disk.qcow2",
+        "blueprint": {}
+    }
+}
+
+It then outputs a json test case as standard output. 
+'''
+
+
+def run_osbuild(manifest, store):
+    osbuild_cmd = ["osbuild", "--store", store, "--json", "-"]
+    result = json.loads(subprocess.check_output(osbuild_cmd, encoding="utf-8", input=json.dumps(manifest)))
+    return result.get("output_id")
+
+
+def main(test_case, store):
+    boot_type = test_case["boot"]["type"]
+    compose_request = test_case["compose-request"]
+
+    blueprint = json.dumps(compose_request["blueprint"])
+
+    pipeline_command = ["go", "run", "./cmd/osbuild-pipeline", "-distro", compose_request["distro"], "-arch", compose_request["arch"], "-image-type", compose_request["output-format"], "-"]
+    test_case["manifest"] = json.loads(subprocess.check_output(pipeline_command, input=blueprint, encoding="utf-8"))
+
+    pipeline_command = ["go", "run", "./cmd/osbuild-pipeline", "-distro", compose_request["distro"], "-arch", compose_request["arch"], "-image-type", compose_request["output-format"], "-rpmmd", "-"]
+    test_case["rpmmd"] = json.loads(subprocess.check_output(pipeline_command, input=blueprint, encoding="utf-8"))
+
+    if boot_type != "nspawn-extract":
+        output_id = run_osbuild(test_case["manifest"], store)
+        image_file = os.path.join(store, "refs", output_id, compose_request["filename"])
+
+        # we don't yet support image-info on directory trees
+        if boot_type == "qemu-extract":
+            fn, ex = os.path.splitext(image_file)
+            if ex == ".xz":
+                with open(fn, "w") as f:
+                    subprocess.run(["xz", "--decompress", "--stdout", image_file], stdout=f)
+                image_file = fn
+        test_case["image-info"] = json.loads(subprocess.check_output(["tools/image-info", image_file]))
+
+    return test_case
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="Generate test cases")
+    parser.add_argument("store", metavar="DIRECTORY", type=os.path.abspath, help="path to the osbuild store")
```

Comment:
I suggest adding `required=True` for the sake of consistency.

Reply:
Since store is a positional argument it is required regardless of having required=True. For consistency, it could be nice to have the store assigned to a flag `--store` as it is in generate-test-cases but since the store is the only argument here I think it is fine as a positional argument.